modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/portlet_url/create_portlet_url.ts

```diff
@@ -95,7 +94,7 @@ export default function createPortletURL(basePortletURL, parameters = {}) {
 	let namespace = '';
 
 	if (Object.entries(parameters).length) {
-		namespace = getPortletNamespace(portletID);
+		namespace = getPortletNamespace(portletID!);
```

Comment:
For sure `portletID` can be `null` here. That's a possible bug somewhere that we haven't yet seen. Thoughts on what should we really do here?

Reply:
Few lines above:
```js
if (Object.entries(parameters).length && !portletID) {
	throw new TypeError(
		'Portlet ID must not be null if parameters are provided'
	);
}
```

Reply:
Generally speaking, I guess we could replace all manual type checks with TS. Unless we want them in runtime, to show some legitimate case in an error log :thinking:

Reply:
So that should be a `||` 😂

Reply:
You see, the thing is that if parameters aren't provided, then `portletID` **can** be null, but then `getPortletNamespace` takes (or could take) non-nullable strings. That's the thing I wanted to raise the attention to. Need to decide how much we lean on the type system. I think the more the better, but maybe in this case `null` is a valid thing to pass to `getPortletNamespace` which doesn't seem to be the case since it will throw a `TypeError` in that case.

Reply:
If parameters are not provided, we never reach `getPortletNamespace`:
```js
if (Object.entries(parameters).length) {
	namespace = getPortletNamespace(portletID);
}
```

With this, the value of `portletID`, that we took it from `basePortletURL`, does not matter.

The bug here might be that we are not checking and namespacing params in `basePortletURL`. But, I'm guessing we are assuming the `basePortletURL` will be used as intended, a base URL, and not as any URL that we are appending out params on.

Reply:
> If parameters are not provided, we never reach getPortletNamespace:

lol, you're totally right... need to get me some better reading glasses 👴 

I wonder why TS is unable to infer that `portletID` can't be `null` at that point. Maybe this could be written in a way that it would make it possible similar to what happens in https://github.com/liferay-frontend/liferay-portal/pull/1379#discussion_r692819269

For this case I'll simply `as string`, then.

Reply:
Maybe putting everything inside one `if`...

```javascript
let namespace = '';

if (Object.entries(parameters).length) {
    if (!portletID) {
        throw new TypeError(
            'Portlet ID must not be null if parameters are provided'
        );
    }
    namespace = getPortletNamespace(portletID);
}
```

Reply:
Yeah, that should do it!